helix-term/src/ui/editor.rs

```diff
@@ -770,6 +770,72 @@ impl Component for EditorView {
 
                 EventResult::Consumed(callback)
             }
+            Event::Mouse(MouseEvent {
+                kind: MouseEventKind::Down(MouseButton::Left),
+                row,
+                column,
+                modifiers,
+                ..
+            }) => {
+                let editor = &mut cx.editor;
+
+                let result = editor.tree.views().find_map(|(view, _focus)| {
+                    if !view.verify_screen_coords(row as usize, column as usize) {
+                        return None;
+                    }
+
+                    Some((
+                        view.pos_at_screen_coords(
+                            &editor.documents[view.doc],
+                            row as usize,
+                            column as usize,
+                        ),
+                        view.id,
+                    ))
+                });
+
+                let tree = &mut editor.tree;
```

Comment:
I don't think this is necessary? Just refer to `editor.tree`

Reply:
I did split it out because of multiple references. Can refer directly if needed by code style.